### ISSUE TITLE
segfault in bcftools csq

### ISSUE THREAD

**Issue by brentp** [samtools/bcftools#524]
[b.vcf.gz](https://github.com/samtools/bcftools/files/632026/b.vcf.gz)
with attached vcf:
```
./bcftools csq --ncsq 132 --local-csq -g Homo_sapiens.GRCh37.82.chr.gff3.gz -f $REF b.vcf.gz
[b.vcf.gz](https://github.com/samtools/bcftools/files/632021/b.vcf.gz)
```
here is the gdb backtrace:
```
#0  __memcpy_avx_unaligned () at ../sysdeps/x86_64/multiarch/memcpy-avx-unaligned.S:268
#1  0x000000000047c867 in memcpy (__len=18446744073709551588, __src=0x6fd4476, __dest=<optimized out>)
    at /usr/include/x86_64-linux-gnu/bits/string3.h:53
#2  kputsn (p=0x6fd4476 "", l=-28, s=0x7fffffffd9b0) at ../htslib/htslib/kstring.h:147
#3  0x000000000048576d in splice_csq_del (ex_end=101719156, ex_beg=101719114, splice=0x7fffffffd950, args=0x753010) at csq.c:1756
#4  splice_csq (args=args@entry=0x753010, splice=splice@entry=0x7fffffffd950, ex_beg=101719114, ex_end=101719156) at csq.c:1985
#5  0x0000000000486184 in hap_init (args=args@entry=0x753010, parent=parent@entry=0x7fffffffdb10, child=child@entry=0x7fffffffdb90, 
    cds=cds@entry=0x662e060, rec=rec@entry=0x6f442e0, ial=ial@entry=1) at csq.c:2024
#6  0x0000000000487625 in test_cds_local (args=args@entry=0x753010, rec=rec@entry=0x6f442e0) at csq.c:3143
#7  0x0000000000488bfe in process (args=args@entry=0x753010, rec_ptr=0x6d03e60) at csq.c:3623
#8  0x0000000000489104 in main_csq (argc=<optimized out>, argv=<optimized out>) at csq.c:3780
#9  0x00007ffff70ea830 in __libc_start_main (main=0x40efb0 <main>, argc=10, argv=0x7fffffffddb8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fffffffdda8) at ../csu/libc-start.c:291
#10 0x000000000040f1e9 in _start ()
```

cc @mcshane @pd3

**Comment by pd3** [comment]
Thank you for the bug report and the test case, this should be now fixed by	02919ec

**Comment by brentp** [comment]
thank you.
